tests/drop_counters/test_drop_counters.py

```diff
@@ -234,36 +403,125 @@ def base_verification(discard_group, pkt, ptfadapter, duthost, combined_counter,
+    testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["neighbor_sniff_ports"])
+
+
+def test_multicast_smac_drop(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields):
+    """
+    @summary: Verify that packet with multicast SMAC is dropped and L2 drop counter incremented
+    """
+    multicast_smac = "01:00:5e:00:01:02"
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+
+    log_pkt_params(dut_iface, dst_mac, multicast_smac, pkt_fields["ipv4_dst"], pkt_fields["ipv4_src"])
+
+    pkt = testutils.simple_tcp_packet(
+        eth_dst=dst_mac, # DUT port
+        eth_src=multicast_smac,
+        ip_src=pkt_fields["ipv4_src"], # PTF source
+        ip_dst=pkt_fields["ipv4_dst"], # VM source
+        tcp_sport=pkt_fields["tcp_sport"],
+        tcp_dport=pkt_fields["tcp_dport"])
+
+    base_verification("L2", pkt, ptfadapter, duthost, ptf_tx_port_id, dut_iface)
+
+    # Verify packets were not egresed the DUT
+    exp_pkt = expected_packet_mask(pkt)
+    testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["neighbor_sniff_ports"])
+
+
+def test_reserved_dmac_drop(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields):
+    """
+    @summary: Verify that packet with reserved DMAC is dropped and L2 drop counter incremented
+    @used_mac_address:
+        01:80:C2:00:00:05 - reserved for future standardization
+        01:80:C2:00:00:08 - provider Bridge group address
+    """
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+    reserved_mac_addr = ["01:80:C2:00:00:05", "01:80:C2:00:00:08"]
+
+    for reserved_dmac in reserved_mac_addr:
+        log_pkt_params(dut_iface, dst_mac, reserved_dmac, pkt_fields["ipv4_dst"], pkt_fields["ipv4_src"])
+        pkt = testutils.simple_tcp_packet(
+            eth_dst=reserved_dmac, # DUT port
+            eth_src=src_mac,
+            ip_src=pkt_fields["ipv4_src"], # PTF source
+            ip_dst=pkt_fields["ipv4_dst"], # VM source
+            tcp_sport=pkt_fields["tcp_sport"],
+            tcp_dport=pkt_fields["tcp_dport"])
+
+        base_verification("L2", pkt, ptfadapter, duthost, ptf_tx_port_id, dut_iface)
+
+        # Verify packets were not egresed the DUT
+        exp_pkt = expected_packet_mask(pkt)
+        testutils.verify_no_packet_any(ptfadapter, exp_pkt, ports=setup["neighbor_sniff_ports"])
+
+
+def test_not_expected_vlan_tag_drop(ptfadapter, duthost, setup, tx_dut_ports, pkt_fields):
+    """
+    @summary: Verify that VLAN tagged packet which VLAN ID does not match ingress port VLAN ID is dropped
+              and L2 drop counter incremented
+    """
+    dut_iface, ptf_tx_port_id, dst_mac, src_mac = get_test_ports_info(ptfadapter, duthost, setup, tx_dut_ports)
+
+    log_pkt_params(dut_iface, dst_mac, src_mac, pkt_fields["ipv4_dst"], pkt_fields["ipv4_src"])
+
+    # Generate unexisted vlan value
+    vlan_id = None
```

Comment:
However unlikely, I think we should try to avoid a case where we keep randomly selecting ints that are in the VLAN, causing the test to run for a long time. One idea:
```suggestion
   upper_bound = max(setup["vlans"])
   vlan_id = [x for x in range(upper_bound) if x not in setup["vlans"]][0]
```

Reply:
Fixed.